JavaScript/JavaScriptSDK/Context/Sample.ts

```diff
@@ -2,6 +2,48 @@
     "use strict";
 
     export class Sample {
-        public sampleRate: string;
+        public sampleRate: number;
+
+        public INT_MAX_VALUE: number = Math.pow(2, 32) - 1;
+
+        constructor(sampleRate: number) {
+            if (sampleRate > 100 || sampleRate < 0) {
+                _InternalLogging.throwInternalUserActionable(LoggingSeverity.CRITICAL, "Sampling rate is out of range (0..100): '" + sampleRate
+                    + "'. Sampling will be disabled, you may be sending too much data which may affect your AI service level.");
+                this.sampleRate = 100;
+            }
+
+            this.sampleRate = sampleRate;
+        }
+
+        public IsSampledIn(envelope: Telemetry.Common.Envelope): boolean {            
```

Comment:
isSampledIn is better - meaning that the package is still in after applying sampling logic. Actually naming is taken from the server-side SDK implementation by Vitaly Filimonov.